server/api/users.js

```diff
@@ -283,7 +285,29 @@ internals.applyRoutes = function (server, next) {
                 }
             };
 
-            User.findByIdAndUpdate(id, update, (err, user) => {
+            const filterById = {
+                'user.id': request.params.id
+            };
+
+            const updateReference = {
+                $set: {
+                    'user.name': request.payload.username
+                }
+            };
+            Async.auto({
+                user: function (done) {
+
+                    User.findByIdAndUpdate(id, update, done);
+                },
+                account: function (done) {
+
+                    Account.findOneAndUpdate(filterById, updateReference, done);
+                },
+                admin: function (done) {
+
+                    Admin.findOneAndUpdate(filterById, updateReference, done);
+                }
+            }, (err, { user }) => {
```

Comment:
Instead of destructuring user from the object being passed in, let's name the argument `results` and use `results.user` where needed.